bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_Node.cpp

```diff
@@ -185,6 +188,26 @@ py::object getObject(Node &n, const std::string &name, const py::kwargs& kwargs)
     return py::none();
 }
 
+void setFieldsFromPythonValues(Base* self, const py::kwargs& dict)
+{
+    // For each argument of the addObject functin we check if this is a
+    // and argument we can do a raw conversion from.
+    // currently, only the LinkPath object is supported.
+    for(auto kv : dict)
+    {
+        if(py::isinstance<LinkPath>(kv.second))
+        {
+            auto* data = self->findData(py::str(kv.first));
+            if(data)
+                BindingBase::SetData(data, py::cast<py::object>(kv.second));
+
+            auto* link = self->findLink(py::str(kv.first));
+            if(link)
+                BindingBase::SetLink(link, py::cast<py::object>(kv.second));
+        }
+    }
```

Comment:
```suggestion
    for(auto [key, value] : dict)
    {
        if(py::isinstance<LinkPath>(value))
        {
            auto* data = self->findData(py::str(key));
            if(data)
                BindingBase::SetData(data, py::cast<py::object>(value));

            auto* link = self->findLink(py::str(key));
            if(link)
                BindingBase::SetLink(link, py::cast<py::object>(value));
        }
    }
```

This suggestion using a structured binding is not mandatory, but I think it's a case where it helps the reading

Reply:
I don't see this as resolved